packages/compass-data-modeling/src/services/export-diagram.tsx

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+import {
+  getNodesBounds,
+  getViewportForBounds,
+  DiagramProvider,
+  Diagram,
+} from '@mongodb-js/diagramming';
+import type {
+  useDiagram,
+  NodeProps,
+  NodeType,
+  EdgeProps,
+  Marker,
+} from '@mongodb-js/diagramming';
+import type { StaticModel } from './data-model-storage';
+import ReactDOM from 'react-dom';
+import { toPng } from 'html-to-image';
+import { rafraf, spacing } from '@mongodb-js/compass-components';
+
+// TODO: Export these methods (and type) from the diagramming package
+type DiagramInstance = ReturnType<typeof useDiagram>;
+function mapNodeToDiagramNode(
+  node: ReturnType<DiagramInstance['getNodes']>[number]
+): NodeProps {
+  const { data, type, ...restOfNode } = node;
+  return {
+    ...restOfNode,
+    ...(data as any), // TODO: Type data (or expose these methods from the diagramming package)
+    type: type as NodeType,
+  };
+}
+function mapEdgeToDiagramEdge(
+  edge: ReturnType<DiagramInstance['getEdges']>[number]
+): EdgeProps {
+  const { markerStart, markerEnd, ...restOfEdge } = edge;
+
+  // The diagramming package only allows string based markers
+  if (typeof markerStart !== 'string' || typeof markerEnd !== 'string') {
+    throw new Error('Unexpected edge with non-string markers');
+  }
+
+  return {
+    ...restOfEdge,
+    markerEnd: markerEnd.replace('end-', '') as Marker,
+    markerStart: markerStart.replace('start-', '') as Marker,
+  };
+}
+
+function moveSvgDefsToViewportElement(
+  container: Element,
+  targetElement: Element
+) {
+  const markerDef = container.querySelector('svg defs');
+  if (!markerDef) {
+    return;
+  }
+  const diagramSvgElements = targetElement.querySelectorAll('svg');
+  diagramSvgElements.forEach((svg) => {
+    const pathsWithMarkers = svg.querySelectorAll(
+      'path[marker-end], path[marker-start]'
+    );
+    if (pathsWithMarkers.length !== 0) {
+      const clonedDef = markerDef.cloneNode(true) as SVGMarkerElement;
+      svg.insertBefore(clonedDef, svg.firstChild);
+    }
+  });
+  markerDef.remove();
+}
+
+export async function exportToPng(
+  fileName: string,
+  containerRef: React.RefObject<HTMLDivElement>,
+  diagram: DiagramInstance
+) {
+  const container = containerRef.current;
+  if (!container) {
+    throw new Error('Container reference is not set');
+  }
+  const dataUri = await getExportPngDataUri(container, diagram);
+  downloadFile(dataUri, fileName, () => {
+    ReactDOM.unmountComponentAtNode(container);
+  });
+}
+
+export function getExportPngDataUri(
+  container: HTMLDivElement,
+  diagram: DiagramInstance
+): Promise<string> {
+  return new Promise<string>((resolve, reject) => {
```

Comment:
Instead of passing container, I would suggest to create container inside this method to make it even more encapsulated

```suggestion
  container: HTMLDivElement,
  diagram: DiagramInstance
): Promise<string> {
  const container = document.createElement('div');
  return new Promise<string>((resolve, reject) => {
```

Reply:
yeah that seems better option.